File: 健录APP-v1.0/fitness-server/src/main/java/com/fitness/server/entity/Workout.java
package com.fitness.server.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;

@TableName("workouts")
public class Workout {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private Long localId;
    private String name;
    private Long date;
    private Long duration;
    private String notes;
    private Long createdAt;
    private Long updatedAt;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }
    public Long getLocalId() { return localId; }
    public void setLocalId(Long localId) { this.localId = localId; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public Long getDate() { return date; }
    public void setDate(Long date) { this.date = date; }
    public Long getDuration() { return duration; }
    public void setDuration(Long duration) { this.duration = duration; }
    public String getNotes() { return notes; }
    public void setNotes(String notes) { this.notes = notes; }
    public Long getCreatedAt() { return createdAt; }
    public void setCreatedAt(Long createdAt) { this.createdAt = createdAt; }
    public Long getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(Long updatedAt) { this.updatedAt = updatedAt; }
}

File: app/src/main/java/com/fitness/training/data/entity/Workout.kt
package com.fitness.training.data.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date

@Entity(tableName = "workouts")
data class Workout(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val userId: Long = 0, // 用户ID
    val name: String = "",
    val date: Long = System.currentTimeMillis(), // 训练日期时间戳
    val duration: Long = 0, // 训练时长（毫秒）
    val notes: String = "", // 训练备注
    val templateId: Long? = null // 关联的训练模板ID
)



